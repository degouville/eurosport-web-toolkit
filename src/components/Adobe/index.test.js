import React from 'react';
import { shallow, mount } from 'enzyme';
import { Adobe } from '../..';

describe('<Adobe /> client side', () => {
  it('renders <Adobe />', () => {
    const component = shallow(<Adobe src="srcUrl" isServerSide={false} />);
    expect(component).toMatchSnapshot();
    component.unmount();
  });

  it('adds the adobe script in the <head>', () => {
    const component = mount(<Adobe src="scriptSourceURL" isServerSide={false} />);
    const adobeScript = global.document.getElementById('adobe-script');
    expect(adobeScript).toBeDefined();
    component.unmount();
  });

  it('sets the right src url', () => {
    const component = mount(<Adobe src="urlToTest" isServerSide={false} />);
    const adobeScript = global.document.getElementById('adobe-script');
    expect(adobeScript.getAttribute('src')).toEqual('urlToTest');
    component.unmount();
  });

  it('calls pageBottom method on _satellite object', () => {
    const satelliteInjector = `_satellite = {}; _satellite.pageBottom = function() {window.document.mockCounter = 2};`;
    const script = document.createElement('script');
    script.innerHTML = satelliteInjector;
    document.body.appendChild(script);
    // eslint-disable-next-line no-unused-vars
    const component = mount(<Adobe src="urlToTest" isServerSide={false} />);
    expect(global.document.mockCounter).toEqual(2);
    component.unmount();
  });
});

describe('<Adobe /> server side', () => {
  it('adds script tag with isServerSide', () => {
    const wrapper = mount(<Adobe src="srcURL" isServerSide />);
    expect(wrapper.exists('#adobe-script')).toEqual(true);
  });
});
