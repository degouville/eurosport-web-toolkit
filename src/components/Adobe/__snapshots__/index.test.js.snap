// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Adobe /> head script client side renders <Adobe /> 1`] = `
<Fragment>
  <ScriptInjector
    async={false}
    id="adobe-script"
    injectPlace="head"
    innerHTML=""
    isServer={false}
    src="srcUrl"
  />
</Fragment>
`;
